crates/uv-resolver/src/pubgrub/dependencies.rs

```diff
@@ -28,6 +29,10 @@ pub(crate) struct PubGrubDependency {
 }
 
 impl PubGrubDependency {
+    pub(crate) fn and_markers(&mut self, marker: &MarkerTree) {
```

Comment:
Yes, but `MarkerTree::and` takes `&mut Self`.

I think either way can work where we avoid cloning `MarkerTree`. This way, it means using `Arc::make_mut`. If we took ownership, it would be `Arc::unwrap_or_clone`. But, in the `make_mut` case where there is only one reference, we don't need to re-create the `Arc`.

How much this matters, I dunno. But I opted for the option that I thought was less costly. (And there are assuredly designs that are even less costly than what I have, but I think it would be a bigger refactor.)